### DISCUSSION TITLE
Interpretation of standard deviation

### DISCUSSION THREAD

**Discussion by max3-2** [lmfit/uncertainties#153] [Q&A] [answered]
Hello,

how can one practically interpret the result of a calculation with uncertainties? I sometimes struggle to understand the results. Assuming a simple case of
```python
from uncertainties import ufloat as uf
a = uf(5, 2)
b = uf(3, 1)
a + b

Out[64]: 8.0+/-2.23606797749979
```
So this obviously corresponds to the linear expansion and resulting `stdev`. But what was assumed as distribution for this `stdev`? Or is it what the user assumed when specifying the initial quantities? 
And if the latter, is there a way to assume **Uniform** distribution, e.g. computing max. bounds, the above would result in
```python
Out[64]: 8.0+/-3
```

**Comment by Naikless** [comment]
As pointed out in the documentation of this package, the type of probability distribution of the uncertain variable doesn't matter when looking at (linear) error propagation and uncertainty budgets, as these are all just defined in terms of standard deviation/uncertainty.

That being said, the typical practical interpretation of standard deviation would be to define a  level of confidence based on its value. E.g. for a normally distributed value, you can assume that with a ~68% probability, your value will fall within the range defined by +- std. However, this is only exactly true for a normal distribution and your level of confidence could vary otherwise.

Taking your above example, the result of `a+b` describes a variable with an average value of 8 and a standard deviation of 2.236. However, it doesn't hold any information about the distribution behind these values. If you assume a normal distribution, the above mentioned 68% level of confidence corresponds to +- std. But if you assume a uniform distribution, a confidence level of ~68% would be defined by `0.68 * a` with `a` being the symmetric bound defined by `a=sqrt(3)*std`. So in this case, the boundaries for a 68% confidence level would be +- 2.644. 

If you want to maintain the information about the kind of distribution within the variable, you can have a look at the [metrolopy](https://github.com/nrc-cnrc/MetroloPy) package which also defines variables with uncertainties but also uses Monte-Carlo methods and convolved probability distributions. 

However, in many practical cases the assumption of a normal distribution is absolutely valid, especially if you have >3 uncertainty variables that contribute the combined uncertainty, even if all of them are uniformly distributed.

I have taken most of this from the [Guide to the expression of uncertainty in measurement (GUM)](https://www.bipm.org/documents/20126/2071204/JCGM_100_2008_E.pdf/cb0ef43f-baa5-11cf-3f85-4dcd86f77bd6), of which `metrolpy` is heavily influenced. 

I would also welcome anyone of the maintainers to correct me if I am spreading half-truths here, since I am also 100% self-taught when it comes to uncertainty calculations.

**Comment by lebigot** [comment, marked as answer]
This is correct.

I would add two points:

1. To be explicit, `uncertainties` doesn't do Monte-Carlo calculations (they are slow). 
2. Its results assume that the functions applied to random variables are well approximated by their linear approximation over the typical range of the variables. This implies that the probability distribution of a function of a single variable has the same distribution up to a linear scaling (so, a uniform distribution remains a uniform distribution). In the same vein, a function of multiple Gaussian random variables also has values that should closely follow a Gaussian distribution (because a linear combination of Gaussian variables is a Gaussian).

Now, to go back to the original question, it is not true that `a+b` is uniform: it is actually triangular, in the same way as the sum of two dice is more likely to give 7 than 12. The results and inputs of `uncertainties` are always standard deviations (so the width of the distribution of `b` is not 2 but is larger).
